File: 백준/BFS/13913.숨바꼭질4/숨바꼭질4.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Deque<Integer> q = new ArrayDeque<>();
        int[] MAP = new int[100001];
        int[] visited = new int[100001];
        int N = sc.nextInt();
        int K = sc.nextInt();
        q.offerFirst(N);
        while (!q.isEmpty()) {
            int now = q.pollFirst();
            if (now == K) {
                System.out.println(MAP[now]);
                int [] answer = new int[MAP[now]+1];
                answer[MAP[now]] = now;
                int tmp = now;
                for (int i = MAP[now]-1; i >= 0; i--) {
                    answer[i] = visited[tmp];
                    tmp = visited[tmp];
                }
                for (int a : answer) {
                    System.out.print(a);
                    System.out.print(' ');
                }
                break;
            }
            int[] nextPositions = {now - 1, now + 1, now * 2};
            for (int i : nextPositions) {
                if (i >= 0 && i <= 100000 && MAP[i] == 0) {
                    q.offerLast(i);
                    MAP[i] = MAP[now] + 1;
                    visited[i] = now;
                }
            }
        }
    }
}

File: 백준/BFS/13913.숨바꼭질4/숨바꼭질4.py
from collections import deque

N, K = map(int, input().split())
MAP = [0]*100001
visited = [0]*100001
q = deque()
q.appendleft(N)
answer = []
while q:
    now = q.popleft()
    if now == K:
        answer.append(now)
        tmp = now
        for i in range(MAP[now]):
            answer.append(visited[tmp])
            tmp = visited[tmp]
        break
    for i in (now-1, now+1, now*2):
        if i >= 0 and i <= 100000 and MAP[i] == 0:
            MAP[i] = MAP[now] + 1
            visited[i] = now
            q.append(i)
    
print(MAP[K])
print(' '.join(map(str, answer[::-1])))